src/java/test/org/apache/zookeeper/test/ClientBase.java

```diff
@@ -100,6 +100,8 @@ public void process(WatchedEvent event) { /* nada */ }
         volatile boolean connected;
         // Set to true when connected to a quorum server.
         volatile boolean syncConnected;
+        // Set to true when connected to a quorum server in read-only mode
+        volatile boolean readOnlyConnected;
```

Comment:
One nit : now there are three variables that encode the connection state and three waitForxxxConnected functions. These could be refactored so we use a single variable and a parameterized function to encode the state and state changes.